Web Inspector: Rename WI.DOMTreeElement.prototype._elementCloseTag

rcaliman-apple opened the pull request:
#### cf37c23ad69cb407e17d9f132de8740e4ae7f54a
<pre>
Web Inspector: Rename WI.DOMTreeElement.prototype._elementCloseTag
<a href="https://bugs.webkit.org/show_bug.cgi?id=299519">https://bugs.webkit.org/show_bug.cgi?id=299519</a>
<a href="https://rdar.apple.com/161318308">rdar://161318308</a>

Reviewed by Devin Rousso.

While investigating <a href="https://commits.webkit.org/300471@main">https://commits.webkit.org/300471@main</a>, the naming of `_elementCloseTag` proved confusing at times.
It is a boolean indicating whether the `WI.DOMTreeElement` represents the closing tag of a DOM element.

There&apos;s a similarly named member, `_closeTagTreeElement`,
which references the actual element of the close tag in the DOM tree outline.

This patch renames `_elementCloseTag` to `isElementCloseTag`, renames the public getter,
and uses the same name across all call sites to make it easier to track across objects.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView.prototype.get selectionPathComponents):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
(WI.DOMTreeElement.prototype.get isElementCloseTag):
(WI.DOMTreeElement.prototype.get expandedChildCount):
(WI.DOMTreeElement.prototype.showChildNode):
(WI.DOMTreeElement.prototype.onpopulate):
(WI.DOMTreeElement.prototype.updateChildren):
(WI.DOMTreeElement.prototype.insertChildElement):
(WI.DOMTreeElement.prototype._updateChildren):
(WI.DOMTreeElement.prototype.onexpand):
(WI.DOMTreeElement.prototype.oncollapse):
(WI.DOMTreeElement.prototype.ondblclick):
(WI.DOMTreeElement.prototype._startEditing):
(WI.DOMTreeElement.prototype._nodeTitleInfo):
(WI.DOMTreeElement.prototype._updatePseudoClassIndicator):
(WI.DOMTreeElement.prototype._createBadges):
(WI.DOMTreeElement.prototype.isCloseTag): Deleted.
* Source/WebInspectorUI/UserInterface/Views/DOMTreeElementPathComponent.js:
(WI.DOMTreeElementPathComponent.prototype.get nextSibling):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.js:
(WI.DOMTreeOutline.prototype.update):
(WI.DOMTreeOutline.prototype.ondelete):
(WI.DOMTreeOutline.prototype.objectForSelection):
(WI.DOMTreeOutline.prototype._ondrop):

Canonical link: <a href="https://commits.webkit.org/300703@main">https://commits.webkit.org/300703@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/543922c62714ece213d1819011a9ee905dd38efa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/123578 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/43293 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/130345 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/75721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/44016 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/51887 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/130345 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/75721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/126531 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/44016 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/130345 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/44016 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/73822 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/44016 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/133036 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/50529 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/51887 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/133036 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/50904 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/133036 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25995 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/33989 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/50383 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/56145 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/49857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/53204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/51532 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -30,14 +30,14 @@
 WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
 {
-    constructor(node, elementCloseTag, {showBadges} = {})
+    constructor(node, {showBadges, isElementCloseTag = false} = {})
```
Style: instead of `= false` we usually do `!!` when it's used
```js
this._isElementCloseTag = !!isElementCloseTag;
```
this way, if a caller provides a non-boolean value it will always be converted to a boolean

rcaliman-apple replied:
Thank you! The reasoning helps clarify why not to rely on the fallback value. I had considered just `undefined`.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -30,14 +30,14 @@
-        this._elementCloseTag = elementCloseTag;
-        this.hasChildren = !elementCloseTag && this._hasVisibleChildren();
+        this._isElementCloseTag = isElementCloseTag;
+        this.hasChildren = !isElementCloseTag && this._hasVisibleChildren();
```
NIT: we generally prefer using the `this._` member variable instead of the argument as the former is after any validations/transformations/etc. (see above) whereas the latter is the raw argument value
```js
this.hasChildren = !this._isElementCloseTag && this._hasVisibleChildren();
```

rcaliman-apple replied:
Makes sense 👍

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -30,14 +30,14 @@
+        this.hasChildren = !isElementCloseTag && this._hasVisibleChildren();
 
-        if (this.representedObject.nodeType() === Node.ELEMENT_NODE && !elementCloseTag)
+        if (this.representedObject.nodeType() === Node.ELEMENT_NODE && !isElementCloseTag)
```
ditto
```js
this._canAddAttributes = this.representedObject.nodeType() === Node.ELEMENT_NODE && !this._isElementCloseTag;
```

in this case it might even be worth just getting rid of `_canAddAttributes` entirely by inlining this in `WI.DOMTreeElement.prototype._startEditing`

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -136,9 +136,9 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
     }
 
-    isCloseTag()
+    get isElementCloseTag()
```
Style: since this is now a "simple" `get` please inline it above
```js
get isElementCloseTag() { return this._isElementCloseTag; }
get statusImageElement() { return this._statusImageElement; }
```

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -315,8 +315,8 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
-        console.assert(!this._elementCloseTag);
-        if (this._elementCloseTag)
+        console.assert(!this.isElementCloseTag);
+        if (this.isElementCloseTag)
```
Style: we normally prefer using the member variable for simple `get` when within the same class to avoid the cost of invoking the function
```js
console.assert(!this._isElementCloseTag);
if (this._isElementCloseTag)
```

ditto below

rcaliman-apple replied:
> to avoid the cost of invoking the function

Thanks for the clarification!

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -474,15 +474,15 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
     }
 
-    insertChildElement(child, index, closingTag)
+    insertChildElement(child, index, isElementCloseTag)
```
can we also make this an options bag like you did in the `constructor`?
```js
insertChildElement(child, index, {isElementCloseTag} = {})
```

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.js:
```diff
@@ -173,18 +173,18 @@ WI.DOMTreeOutline = class DOMTreeOutline extends WI.TreeOutline
         this.removeChildren();
 
-        const elementCloseTag = false;
+        const isElementCloseTag = false;
```
there's no need for this anymore since the default value is `false` anyways

we only had it before since it was a required argument instead of optional

rcaliman-apple replied:
>we only had it before since it was a required argument instead of optional

That's right. I'll remove it.

rcaliman-apple committed:
Web Inspector: Rename WI.DOMTreeElement.prototype._elementCloseTag
https://bugs.webkit.org/show_bug.cgi?id=299519
rdar://161318308

Reviewed by Devin Rousso.

While investigating https://commits.webkit.org/300471@main, the naming of `_elementCloseTag` proved confusing at times.
It is a boolean indicating whether the `WI.DOMTreeElement` represents the closing tag of a DOM element.

There's a similarly named member, `_closeTagTreeElement`,
which references the actual element of the close tag in the DOM tree outline.

This patch renames `_elementCloseTag` to `isElementCloseTag`, renames the public getter,
and uses the same name across all call sites to make it easier to track across objects.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView.prototype.get selectionPathComponents):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
(WI.DOMTreeElement.prototype.get isElementCloseTag):
(WI.DOMTreeElement.prototype.get expandedChildCount):
(WI.DOMTreeElement.prototype.showChildNode):
(WI.DOMTreeElement.prototype.onpopulate):
(WI.DOMTreeElement.prototype.updateChildren):
(WI.DOMTreeElement.prototype.insertChildElement):
(WI.DOMTreeElement.prototype._updateChildren):
(WI.DOMTreeElement.prototype.onexpand):
(WI.DOMTreeElement.prototype.oncollapse):
(WI.DOMTreeElement.prototype.ondblclick):
(WI.DOMTreeElement.prototype._startEditing):
(WI.DOMTreeElement.prototype._nodeTitleInfo):
(WI.DOMTreeElement.prototype._updatePseudoClassIndicator):
(WI.DOMTreeElement.prototype._createBadges):
(WI.DOMTreeElement.prototype.isCloseTag): Deleted.
* Source/WebInspectorUI/UserInterface/Views/DOMTreeElementPathComponent.js:
(WI.DOMTreeElementPathComponent.prototype.get nextSibling):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.js:
(WI.DOMTreeOutline.prototype.update):
(WI.DOMTreeOutline.prototype.ondelete):
(WI.DOMTreeOutline.prototype.objectForSelection):
(WI.DOMTreeOutline.prototype._ondrop):

Canonical link: https://commits.webkit.org/300703@main
